File: 精進/2020:01/fusing_slimes.py
# coding: utf-8
# Your code here!
import itertools
n=int(input())
x = list(map(int, input().split()))
last = x[-1]
p_list = list(itertools.permutations(x,n))
test = []
for i in range(n):
    if p_list[i][-1] == last:
        test.append(p_list[i])
sum = 0
for l in range(len(test)):
    for i in range(n-1):
        sum += abs(test[l][i]-test[l][i+1])
print(sum )

File: 精進/2020:01/fusing_slimes.cpp
#include <bits/stdc++.h>
using namespace std;

int kaijyo(int k) {
    if (k ==1) return 1;
    else {
        return k*kaijyo(k-1);
    }
}
int main(void){
    // Your code here!
    int n;
    cin >> n;
    vector<long long> x(n+1);
    for ( int i = 1 ; i <= n ; i++) {
        cin >> x.at(i);
    }
    long long  sum = 0;
    for (int i = 1; i < n;i++) {
        for (int  j =1; j <= i;j++) {
            sum+=(x.at(i+1)-x.at(i))*kaijyo(n-1)/j;
        }
    }
    int result;
    result = sum % (1000000000+7);
    cout << result << endl;
}